### PR TITLE
Enhance the env-function documentation

### PR DESCRIPTION
**Pull request by weskamm**
This adds a hint for the usage of the env function in dynamic symbolizers (online resource in external graphic).
In that case one needs to put single quotes around the first argument to make things work.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/user/source/filter/function_reference.rst [hunk 1]
```diff
@@ -108,7 +108,9 @@ Environment function
 This function returns the value of environment variables
 defined in various contexts.
 WMS GetMap automatically defines somes variables :ref:`SLD rendering <sld_variable_substitution>`,
-while others can be provided using the ``env`` request parameter.
+while others can be provided using the ``env`` request parameter. When using this function for
+dynamic symbolizers make sure to put the argument of this function in single quotes,
+e.g. ``${env('size', 20)}``.
```
Comment (aaime) [inline doc/en/user/source/filter/function_reference.rst:113]: It has to be put in single quotes everywhere. An unquoted string in CQL is an attribute name reference (no matter where, it's not just dynamic symbolizes). It's a also valid usage, it would mean the env var name to be looked after is contained in the attribute value.

Comment (weskamm) [reply doc/en/user/source/filter/function_reference.rst:113]: Yeah but that may not be clear to everyone.
Looking at the documentation here
https://docs.geoserver.org/stable/en/user/styling/sld/extensions/pointsymbols.html#dynamic-symbolizers
especially in the last code block at the bottom one could get the idea that function arguments may be given without quotes. Feel free to close this PR as it may just be me who got confused here :-)

Comment (aaime) [reply doc/en/user/source/filter/function_reference.rst:113]: At least, I'd reword it to avoid mentioning the dynamic symbolizer case (there is nothing specific to it). Maybe just add an example or two, that should be enough?

Comment (weskamm) [reply doc/en/user/source/filter/function_reference.rst]: I have added examples and reworded a bit. The dynamic symbolizer is still there as an example, as i dont know other ways of calling the env function besides the `classic` way

Comment (aaime) [reply doc/en/user/source/filter/function_reference.rst]: Looks good, thanks! Merged.
